pkg/asset/installconfig/azure/validation.go

```diff
@@ -146,11 +147,46 @@ func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceTyp
 		}
 	}
 
-	// The UltraSSDAvailable capability might not be present at all, in which case it must assumed to be false
-	val, ok = capabilities["UltraSSDAvailable"]
-	if ok {
-		ultraSSDAvailable := strings.EqualFold(val, "True")
-		if ultraSSDEnabled && !ultraSSDAvailable {
+	if ultraSSDEnabled {
```

Comment:
Perhaps I'm overloading this PR, but IMHO this function is doing too much, is hard to read and could be refactored to break down each check into its own function, roughly following this pattern:

https://github.com/openshift/installer/blob/0a8f60ffca5edae87a64b0f81d56e352bf6caaad/pkg/asset/installconfig/azure/validation.go#L40-L44

 So, in this case we could have something like:

```go
allErrs := field.ErrorList{} 
  
 allErrs = append(allErrs, validateCPUs(client, fieldPath, capabilities)...) 
 allErrs = append(allErrs, validateMemory(client, fieldPath, capabilities)...)
allErrs = append(allErrs, validateUltraSSD(client, fieldPath, capabilities, zones)...) 
```
This is a simplified example, but WDYT?

Reply:
I added a new commit implementing this. LMK what you think.